zirgen/Dialect/BigInt/Precompiles.md

```diff
@@ -0,0 +1,106 @@
+# Creating a precompile using the BigInt2 system
+
+Creating a new accelerator using BigInt2 requires a series of related changes across files in two repositories. This is the process you should follow if you want to create a new bigint2 precompile and make it available within the zkVM.
+
+To keep the demo simple, let's imagine that we want to create a trivial accelerator which sums two 128-bit numbers, which we'll call "add128".
+
+
+First, create files under [`zirgen/circuit/bigint/`] which will hold the accelerator builder function. We might call them `add128.h` and `add128.cpp`, declaring and then defining a function like this:
+
+```
+void genAdd128(mlir::OpBuilder& builder, mlir::Location loc) {
```

Comment:
@tzerrell i wonder whether there's a reference we could add here that would enable ppl to teach themselves how to write these programs...

- somewhere in the MLIR docs that would be helpful orientation? 
- a code comment to clarify some of the details of the dialect? 
- somewhere in the overview md doc in the `Dialect/` directory?

Reply:
I think this would be a substantial additional bit of work. Nothing _too_ hard, but I don't think that the mainline MLIR tutorials really cover OpBuilder, and its [reference docs](https://mlir.llvm.org/doxygen/classmlir_1_1OpBuilder.html) would not be helpful here. That means we'd have to write our own tutorial -- it's more than just a comment or even a paragraph.

This might be worth doing, and it's probably not that big a project, but it's its own thing.

Reply:
let's not build a tutorial right now -- i thinking more along the lines of "go look at the RSA example and the underlying struct definitions and good luck"